### PR TITLE
Fixes 1171:  Add react tests for contentListFilters

### PR DESCRIPTION
**Pull request by Andrewgdewar**
Updated react tests to cover the contentListFilters component.

Additional Changes: 
- Fixed a bug within useDebounce that was causing rerenders due to a pendo.js timer.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-533

**Comment by jlsherrill** [comment]
I think this jira ticket is incorrect?

**Review by jlsherrill — APPROVED** [review]
simple enough!

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-1171

File: src/services/useDebounce.tsx [hunk 1]
```diff
@@ -1,14 +1,18 @@
+import { isEqual } from 'lodash';
 import { useEffect, useState } from 'react';
 
 function useDebounce<T>(value: T, delay?: number): T {
   const [debouncedValue, setDebouncedValue] = useState<T>(value);
 
   useEffect(() => {
-    const timer = setTimeout(() => setDebouncedValue(value), delay || 500);
+    // We need to make sure that we compare-deep here as the default useEffect deps do not.
```
Comment (swadeley) [inline src/services/useDebounce.tsx:8]: Re "compare-deep": Why the hyphen? Do you mean "deps"?

Comment (swadeley) [reply src/services/useDebounce.tsx:8]: OK, I see the concept used in JavaScript. I'll so no more about he hyphen. Thank you.

**Pull request merged**